### PR TITLE
test: add unit tests for canAbsorbCommit

### PR DESCRIPTION
**Pull request by brychanrobot**
🎯 **What:** The testing gap addressed
This PR addresses missing unit tests for the `canAbsorbCommit` function in `src/utils/jj-utils.ts`, which determines whether a commit has any mutable parents and thus can absorb changes.

📊 **Coverage:** What scenarios are now tested
The new tests cover the following scenarios:
- Missing `parents` array on the commit object
- Empty `parents` array
- All parents in the array being immutable
- At least one parent in the array being mutable

✨ **Result:** The improvement in test coverage
We now have comprehensive test coverage for `canAbsorbCommit`, ensuring that its behavior remains reliable and that potential refactors can be performed confidently.

---
*PR created automatically by Jules for task [14444721188006714939](https://jules.google.com/task/14444721188006714939) started by @brychanrobot*

## Summary by Sourcery

Tests:
- Add tests covering missing parents, empty parents array, all-immutable parents, and at-least-one-mutable parent scenarios for canAbsorbCommit.

### CODE REVIEW

File: src/test/jj-utils.test.ts [hunk 1]
```diff
@@ -172,6 +173,26 @@ describe('JJ Utils', () => {
...
     });
 
+    describe('canAbsorbCommit', () => {
+        it('should return false if parents array is missing', () => {
+            expect(canAbsorbCommit({})).toBe(false);
+        });
+
+        it('should return false if all parents are immutable', () => {
+            expect(canAbsorbCommit({ parents: [{ is_immutable: true }] })).toBe(false);
+            expect(canAbsorbCommit({ parents: [{ is_immutable: true }, { is_immutable: true }] })).toBe(false);
```
[the remark this answers is not in the document]
Comment (brychanrobot) [reply src/test/jj-utils.test.ts]: Done
